lib/Partitioner/Partitioner.cpp

```diff
@@ -1348,27 +1367,97 @@ llvm::Error Partitioner::Partition(CompilationContext &cctx) {
     dumpDAG("DAG.dot");
   }
 
-  int i = 0;
   for (Function *subF : funcList) {
     (void)subF;
-    if (logPartition) {
-      LOG(INFO) << "\t Partition " << i << ":\n"
-                << "\t\t Name :\t" << subF->getName().str() << "\n"
-                << "\t\t BackendKind :\t"
-                << mapping.getPartitionBackendName(subF) << "\n"
-                << "\t\t Memory :\t"
-                << mapping.getGraphMemInfo(subF).getTotalMemSize() << "\n"
-                << "\t\t LogicalDeviceIDs :\t"
-                << mapping.getLogicalDeviceIDList(subF)[0] << "\n";
-    }
     if (dumpPartition) {
       subF->dumpDAG("partitionLogicalID" +
                     std::to_string(mapping.getLogicalDeviceIDList(subF)[0]) +
                     "__" + subF->getFilename() + "__" +
                     mapping.getPartitionBackendName(subF) + ".dot");
     }
-    i++;
     assert(subF->verify() && "Conversion led to invalid function");
   }
+  if (logPartition) {
+    dumpPartitionInfo(mapping);
+  }
+  return llvm::Error::success();
+}
+
+llvm::Error Partitioner::PartitionFromConfig() {
+  Function *F = module_->getFunction(partitionConfig_.funcName);
+  RETURN_ERR_IF_NOT(F, "Can't find the function in current module.");
+
+  NodeToFunctionMap partitionMap;
+  std::vector<Function *> funcList;
+  std::unordered_set<size_t> unUsed;
+  std::vector<NodesSet> nodesSets;
```

Comment:
```suggestion
  std::vector<NodesSet> nodesSets(partitionConfig_.numOfPartitions);
```
Then you won't need `nodesSets.push_back(NodesSet{});` later.